File: Sources/App/Models/ArtistImage.swift
//
//  ArtistImage.swift
//  FinchServer
//
//  Created by Stefan Klein Nulent on 10/06/2025.
//

import Vapor
import Fluent

final class ArtistImage: Model, @unchecked Sendable {
    
    // MARK: - Types
    
    enum ImageType: String, Codable {
        case background
        case image
        case thumbnail
        case musicLogo
    }
    
    
    
    // MARK: - Constants
    
    static let schema = "artist_images"
    
    
    
    // MARK: - Properties
    
    @ID(custom: .id, generatedBy: .database)
    var id: Int?
    
    @Field(key: "filename")
    var filename: String
    
    @Enum(key: "image_type")
    var type: ImageType
    
    @Parent(key: "artist_id")
    var artist: Artist
    
    
    
    // MARK: - Construction
    
    init(_ type: ImageType, filename: String) {
        self.type = type
        self.filename = filename
    }
    
    init() {}
}

File: FinchServer/Database/ArtistImage.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FinchServer.Database;

[Table("artist_image")]
public class ArtistImage {
    
    // - Properties
    
    [Key]
    public int Id { get; init; }
    
    [MaxLength(255)] 
    public required string FileName { get; init; }
    
    public ImageType ImageType { get; init; }
    public Artist Artist { get; set; } = null!;
}